src/core/dash_internal.h

```diff
@@ -1587,6 +1587,11 @@ auto Segment<Key, Value, Policy>::BumpUp(uint8_t bid, SlotId slot, Hash_t key_ha
   constexpr unsigned kLastSlot = kNumSlots - 1;
   assert(swapb.GetBusy() & (1 << kLastSlot));
 
+  // Don't move sticky items back to the stash because they're not evictable
+  if (swapb.value[kLastSlot].IsSticky()) {
+    return Iterator{bid, slot};
+  }
```

Comment:
In addition, the correct behavior is more complicated than what you implemented.

if we have a sticky entry A and the next one B, which we bump up, I would expect that we won't swap A with B even if both reside in the normal bucket - because this would create drifting of sticky entries to end of the bucket.
And of course, we should not swap A with B if B is in the stash bucket.
now, you have only the stash case scenario. 

even if we focus only on it, i suspect that it still has a bug because at line 1583 we perform destructive `UnsetStashPtr` call. You need to revert it in case you give up.

For the protocol, I think that BumpUp should do more than just give up on bumping up elements if they need to swap with a sticky item but it can definitely wait for another iteration. I would still add a TODO that we should consider finding the first item that is not sticky in that bucket.